theatre/core/src/propTypes/index.ts

```diff
@@ -491,34 +529,41 @@ export function stringLiteral<Opts extends {[key in string]: string}>(
     valueType: null as $IntentionalAny,
     as: opts?.as ?? 'menu',
     label: opts?.label,
-    sanitize(value: unknown) {
-      if (typeof value !== 'string') return undefined
-      if (Object.hasOwnProperty.call(options, value)) {
-        return value as $IntentionalAny
+    interpolate: opts?.interpolate ?? leftInterpolate,
+    deserialize(json: unknown): undefined | Extract<keyof Opts, string> {
+      if (typeof json !== 'string') return undefined
+      if (Object.prototype.hasOwnProperty.call(options, json)) {
+        return json as $IntentionalAny
       } else {
         return undefined
       }
     },
-    interpolate: leftInterpolate,
   }
 }
 
 export type Sanitizer<T> = (value: unknown) => T | undefined
 export type Interpolator<T> = (left: T, right: T, progression: number) => T
 
```

Comment:
Fixed here: f3dfb7cedb03dea0fa7c629e32eaa31ba20c10b3

> instead of left / right, we called it t0 and t1, so the function would read interpolate<V>(t0: V, t1: V, t: number), but we had a few different versions like (tn1, t1, t) as well for t: -1 to 1

Curious what the use-case was?

My own reasons for going for a `[0...1]` range was that the progression could imply time, but also space (scroll) or something else. Also, the math for `[0..1]` is a bit easier, especially for end-users who may write their own interpolation functions, for `t.string()` for example.